lib/rules/named-import.js

```diff
@@ -0,0 +1,238 @@
+
+      return (fixer) => fixer.replaceTextRange(range, getImportDeclarationString());
+    }
+
+    function shouldUpdateImportStatement() {
+      const currentSpecifiers = importDeclaration.specifiers
+        .filter((specifier) => specifier.type !== 'ImportDefaultSpecifier')
+        .map((specifier) => specifier.imported.name);
+
+      return specifiers.some((spec) => currentSpecifiers.indexOf(spec) === -1)
+        || currentSpecifiers.some((spec) => isNameOfType(spec, 'property'));
+    }
+
+    function getFixer(name, range, newType) {
+      const newText = `${newType === 'property' ? 'React.' : ''}${name}`;
+
+      return (fixer) => fixer.replaceTextRange(range, newText);
+    }
+
+    function getReferenceIdentifiers(node) {
+      const variables = context.getDeclaredVariables(node);
+      return variables[0].references.map((reference) => reference.identifier);
+    }
+
+    function report(node, newType, _name, _range) {
+      const name = _name || node.name;
+      const range = _range || node.range;
+
+      if (newType === 'import') {
+        specifiers.push(name);
+      }
+
+      context.report({
+        node,
+        messageId: newType === 'property' ? 'usePropertyAccessor' : 'useNamedImport',
+        data: {name},
+        fix: getFixer(name, range, newType)
+      });
+    }
+
+    function getTypeNamespace(node) {
+      if (node.type === 'TSQualifiedName') {
+        if (node.left) {
+          return getTypeNamespace(node.left);
+        }
+      }
+
+      if (node.type === 'Identifier') {
+        return node.name;
+      }
+    }
+
+    function getTypeProperties(node) {
+      if (!node) {
+        return;
+      }
+      if (node.type === 'TSQualifiedName') {
+        const namespace = getTypeNamespace(node);
+        return {
+          namespace,
+          name: node.right.name
+        };
+      }
+
+      if (node.type === 'Identifier') {
+        return {
+          namespace: null,
+          name: node.name
+        };
+      }
+    }
+
+    const typeSelector = (node) => {
+      const tsType = getTypeProperties(node);
+      if (!tsType) { return; }
+
+      let type;
+
+      if (tsType.namespace === 'React') {
+        type = 'import';
+      } else if (propertySpecifiers.indexOf(tsType.name) > -1) {
+        type = 'property';
+      }
+
+      if (!type) { return; }
+
+      if (isNameOfType(tsType.name, type)) {
+        report(
+          node,
+          type,
+          tsType.name
+        );
+      }
+    };
+
+    return {
+      'Program:exit'(node) {
+        if (shouldUpdateImportStatement()) {
+          context.report({
+            node,
+            messageId: 'fixImportStatement',
+            fix: fixImportDeclaration()
+          });
+        }
+
+        importDeclaration = null;
+        specifiers = [];
+        propertySpecifiers = [];
+      },
+      TSTypeReference: (node) => typeSelector(node.typeName),
+      'ImportDeclaration[source.value=\'react\']'(node) {
```

Comment:
this needs to use the pragma setting, not hardcode "react"

Reply:
`import React from "react"`

the first "React" would be the pragma right? This hardcoded string stands for the package name, which is always "react". Or am I mistaken?

Reply:
oh hm, that's true. Looks like we currently just hardcode the react package name.